Civi/Api4/Generic/AbstractGetAction.php

```diff
@@ -34,13 +34,13 @@ public function selectRowCount() {
   /**
    * Adds field defaults to the where clause.
    *
-   * Note: it will skip adding field defaults when fetching records by id,
+   * Note: it will skip adding field defaults when fetching records by name or id,
    * or if that field has already been added to the where clause.
    *
    * @throws \CRM_Core_Exception
    */
   public function setDefaultWhereClause() {
-    if (!$this->_itemsToGet('id')) {
```

Comment:
Is this the best fix - feels like it could affect a *lot* of other API queries?

Reply:
@ufundo I would say yes, under the circumstances it is the best fix. Fetching a record by `name` is basically the same as fetching it by `id` so it should behave the same. I can't see this negatively affecting other api calls, but it should prevent some future headaches.

I say 'under the circumstances' because I think the *best* fix would be to go back in time 5 years and not put in these hard-coded default where clauses in the first place. But short of that...

Reply:
Sorry I don't have the context. I'm also confused at what this defaultWhere clause is. Automatically exclude any records that don't have default value for any fields that have a default value? :thinking: 

Also just to sense check, `_itemsToGet('name')` **doesn't** get populated for queries a where clause like `['name', 'CONTAINS', 'a'] `?

Reply:
Whoops, I spoke too soon. Test failure says it's more complicated. Hang on I have an idea to do something smarter here & check unique indices instead of fieldname guesswork.